companion/src/mdichild.cpp

```diff
@@ -320,9 +338,9 @@ bool MdiChild::hasPasteData() const
   return mimeData->hasFormat("application/x-companion");
 }
 
-bool MdiChild::hasSelection() const
+bool MdiChild::hasModelSelected()
 {
-  return ui->modelsList->selectionModel()->hasSelection();
+  return isModel(ui->modelsList->currentIndex());
```

Comment:
Hmm, it's undocumented what returns if the `currentIndex()` is invalid (that is, no current selection).  Can you verify?  It's probably -1 or something similarly harmless, but...